src/inmanta/util.py

```diff
@@ -699,3 +701,24 @@ async def __aenter__(self) -> T:
 
     async def __aexit__(self, *excinfo: object) -> None:
         pass
+
+
+async def join_threadpools(threadpools: List[ThreadPoolExecutor]) -> None:
+    # idea borrowed from BaseEventLoop.shutdown_default_executor
+    loop = asyncio.get_event_loop()
+    future = loop.create_future()
+
+    def join() -> None:
+        for threadpool in threadpools:
+            try:
+                threadpool.shutdown(wait=True)
+            except Exception:
+                LOGGER.exception("Exception during threadpool shutdown")
+        loop.call_soon_threadsafe(future.set_result, None)
+
+    thread = threading.Thread(target=join)
```

Comment:
Do I understand correctly that we start a new thread outside of the threadpool to wait (blocking) for shutdown of the threadloop, thereby allowing the event loop(s?) within the threadpool to continue? It is a lot nicer than what we came up with.

Reply:
You understand correctly. 

It was my alternative idea, only that my idea was to start a threadpool, which is a bit overkill.

Reply:
I think I completely misunderstood your idea